docs/nsm_api.md

```diff
@@ -0,0 +1,445 @@
+Network Service Mesh (NSM) API
+
+## Executive summary
+
+Based on the discussions on NSM IRC channel on September 11th/12th and in the issue [https://github.com/ligato/networkservicemesh/issues/283](https://github.com/ligato/networkservicemesh/issues/283). This document describes API calls between different components in NSM enabled kubernetes cluster. Each type of API is provided with proto file definition and brief description of parameters expected and returned. Developers of NFVi who plan to leverage NSM for their applications, are urged to thoroughly review these API calls to make sure that expected and returned parameters do fully cover their applications&#39; needs.
+
+
+
+## List of Network Service Mesh Components
+
+### Network Service Mesh Components in the Abstract
+
+Current Network Service Mesh is highly focused on Network Service Mesh within the context of a Kubernetes Cluster.  Most of this document will work within that context.
+Network Service Mesh concepts are highly genericizable.
+
+![Network Service Mesh Abstract Concepts](./images/NSM%20Diagrams%20for%20Arch%20Docs.png)
+<dl>
```

Comment:
Rather than using raw HTML, I think you should modify this so it uses standard Markdown syntax. [1] is an example, but there are others. This makes it easier to read the raw file itself.

[1] https://daringfireball.net/projects/markdown/syntax

Reply:
I'm 100% for markdown syntax... but there is no markdown syntax for definition lists :(